### ISSUE TITLE
[Bug]: DetailsList - Grouped: Unnecessary column markup announced by screen reader

### ISSUE THREAD

**Issue by vinfol** [microsoft/fluentui#33565]
### Package

react

### Package version

8.119.2

### React version

18

### Environment

```shell
Chrome: Version 131.0.6778.205 (Official Build) (64-bit)
Windows 11
```

### Current Behavior

When using DetailsList groups prop, and set the "collapseAllVisibility" to hidden, the narrator announce wrong column number.
For example:

Narrator is announcing that there are 4 columns instead of 3. Name and Color are announced as 2 of 4 and 3 of 4 (instead of 2 of 3 and 3 of 3)

### Expected Behavior

The screent reader should not announce column markup for first column as there is no data in the first column so there should not be any column markup . The column with no content should be removed.

### Reproduction

https://codepen.io/nadavrosenthal/pen/Exvqoxp

### Steps to reproduce

This issue can repro Fluent UI portal:
https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/basic

Open a screen reader, such Narrator
Use keyboard to navigate to the DetailsList - Grouped
The screen reader read wrong info here, there only three columns, but screen read four.
Seems there have a hidden gridcell
![Image](https://github.com/user-attachments/assets/194fd747-3c53-4d89-b5c8-d6fd94250fde)

There have a similar issue, but closed:
https://github.com/microsoft/fluentui/issues/18657

### Are you reporting an Accessibility issue?

yes

### Suggested severity

Urgent - No workaround and Products/sites are affected

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.
